### PR TITLE
Refactor: docker handler and project class

### PR DESCRIPTION
**Pull request by andxu**
2. add java project class to common lib with should be shared
3. fix typo: 'UNKNOW'

### CODE REVIEW

File: azure-tools-common/src/main/java/com/microsoft/azure/common/project/JavaProject.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.lang3.StringUtils;
+
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.List;
+
+public class JavaProject implements IProject {
+    private String projectName;
```
Comment (Flanker32) [inline azure-tools-common/src/main/java/com/microsoft/azure/common/project/JavaProject.java]: checkstlye issue

File: azure-tools-common/src/main/java/com/microsoft/azure/maven/function/handlers/runtime/DockerFunctionRuntimeHandler.java [hunk 2]
```diff
@@ -79,13 +69,13 @@ protected DockerFunctionRuntimeHandler(Builder builder) {
...
-                .withAppSetting(APP_SETTING_WEBSITES_ENABLE_APP_SERVICE_STORAGE, APP_SETTING_WEBSITES_ENABLE_APP_SERVICE_STORAGE_VALUE)
-                .withAppSetting(APP_SETTING_FUNCTION_APP_EDIT_MODE, APP_SETTING_FUNCTION_APP_EDIT_MODE_VALUE);
+                .withAppSetting(APP_SETTING_WEBSITES_ENABLE_APP_SERVICE_STORAGE, APP_SETTING_DISABLE_WEBSITES_APP_SERVICE_STORAGE)
+                .withAppSetting(APP_SETTING_FUNCTION_APP_EDIT_MODE, APP_SETTING_FUNCTION_APP_EDIT_MODE_READONLY);
     }
 
     @Override
     public FunctionApp.Update updateAppRuntime(FunctionApp app) throws AzureExecutionException {
-        final MavenDockerCredentialProvider provider = MavenDockerCredentialProvider.fromMavenSettings(settings, serverId);
+        final IDockerCredentialProvider provider = StringUtils.isNotBlank(this.serverId) ? this.dockerCredentialProvider : null;
```
Comment (Flanker32) [inline azure-tools-common/src/main/java/com/microsoft/azure/maven/function/handlers/runtime/DockerFunctionRuntimeHandler.java]: Why we add a `StringUtils.isNotBlank(this.serverId)` check here? As `DockerFunctionRuntimeHandler` should not know the implementation of CredentialProvider

Comment (andxu) [reply azure-tools-common/src/main/java/com/microsoft/azure/maven/function/handlers/runtime/DockerFunctionRuntimeHandler.java]: Done

File: azure-tools-common/src/main/java/com/microsoft/azure/common/project/JavaProject.java (hunk 1 above)
Comment (andxu) [reply azure-tools-common/src/main/java/com/microsoft/azure/common/project/JavaProject.java]: Done

**Pull request merged**